engine/src/main/java/org/entando/entando/web/common/interceptor/EntandoOauth2Interceptor.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.oltu.oauth2.common.message.types.ParameterStyle;
+import org.apache.oltu.oauth2.rs.request.OAuthAccessResourceRequest;
+import org.entando.entando.aps.system.exception.RestServerError;
+import org.entando.entando.aps.system.services.oauth2.IApiOAuth2TokenManager;
+import org.entando.entando.aps.system.services.oauth2.model.OAuth2Token;
+import org.entando.entando.web.common.annotation.RestAccessControl;
+import org.entando.entando.web.common.exceptions.EntandoAuthorizationException;
+import org.entando.entando.web.common.exceptions.EntandoTokenException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+/**
+ *
+ * @author paddeo
+ */
+public class EntandoOauth2Interceptor extends HandlerInterceptorAdapter {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private IApiOAuth2TokenManager oAuth2TokenManager;
+
+    @Autowired
+    private IAuthenticationProviderManager authenticationProviderManager;
+
+    @Autowired
+    private IAuthorizationManager authorizationManager;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        if (handler instanceof HandlerMethod) {
+            HandlerMethod method = (HandlerMethod) handler;
+            if (method.getMethod().isAnnotationPresent(RequestMapping.class)) {
+                RestAccessControl rqm = method.getMethodAnnotation(RestAccessControl.class);
+                String permission = rqm.permission();
+                this.extractOAuthParameters(request, permission);
+            }
+        }
+        return true;
+    }
+
+    protected void extractOAuthParameters(HttpServletRequest request, String permission) {
+        try {
+            logger.info("Permission required: {}", permission);
+            OAuthAccessResourceRequest requestMessage = new OAuthAccessResourceRequest(request, ParameterStyle.HEADER);
+
+            String accessToken = requestMessage.getAccessToken();
+            if (StringUtils.isBlank(accessToken)) {
+                throw new EntandoTokenException("no access token found", request, null);
+            }
+
+            final OAuth2Token token = oAuth2TokenManager.getApiOAuth2Token(accessToken);
+            this.validateToken(request, accessToken, token);
+
+            String username = token.getClientId();
+            this.checkAuthorization(username, permission, request);
+
+        } catch (OAuthSystemException | ApsSystemException | OAuthProblemException ex) {
+            logger.error("System exception {}", ex.getMessage());
+            throw new RestServerError("error parsing OAuth parameters", ex);
```

Comment:
This doesn't match up with the checkAuthorization messaging. If a User  isn't authorized an exception is thrown but here we are saying "error parsing parameters"

Reply:
Those exceptions (but not ApsSystemException) depend on the library we are using (Oltu): if you are wrong adding the token to the request, the library class responsible for request management raises one of them